pull_request_template.md

```diff
@@ -0,0 +1,26 @@
+## Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context – why has this been changed/fixed.
+
+References: [TICKET-ID](url), [TICKET-ID](url), …
+
+## How has this been tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can verify the changes. Please describe whether or not you implemented automated tests.
+
+## Things to pay attention to during code review
+
+Please describe parts of the change that require extra attention during code review, for example:
+
+- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
+- Etc.
+
+## Checklist
+
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code in hard-to-understand areas, if any
+- [ ] I have made needed changes to the README
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
```

Comment:
I think I'd like to change this line about testing and move it up to be number two in the list, since it's so important. What do you think about something like:
> - [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
> - [ ] I have added regression tests, if the PR fixes a bug

Reply:
Also, as a third list item, I think we should add something about application observability:
> - [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging

Reply:
Updated to include your suggestions 🙂